### PR TITLE
ci(jenkins): decouple post-release steps in the rake release step

### PR DESCRIPTION
**Pull request by v1v**
## What does this pull request do?

- Update docs
- Create rake task to syncup the branches when a release happens
- Notify if the post-release step failed and keep moving forward. 
- Send an email if the syncup failed.

 
## Why is it important?

Avoid fail fast when the syncup between merges failed, this will help to release the new version of the opbeans.

## Related issues

Closes https://github.com/elastic/apm-agent-ruby/issues/714

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -48,4 +48,5 @@ To release a new version:
     1. Tag the current commit as new version.
     2. Push the tag to GitHub.
     3. Build the gem and upload to Rubygems (local user needs to be signed in and authorized.)
-    4. Update `2.x` branch to be at released commit and push it to GitHub.
+1. Run `rake syncup`. This will...
```
Comment (mikker) [inline CONTRIBUTING.md]:
```suggestion
1. Run `rake release:update_branch`. This will...
```

File: Rakefile [hunk 2]
```diff
@@ -2,7 +2,10 @@
 
 require 'bundler/gem_tasks'
 
-Rake::Task[:release].enhance do
+desc """Post release action:
+Update `3.x` branch to be at released commit and push it to GitHub.
+"""
+task :syncup => :environment do
```
Comment (mikker) [inline Rakefile]:
```suggestion
namespace :release do
  task :update_branch do
```

Comment (mikker) [reply Rakefile]: I don't think `:environment` is needed here or if we even have such a task in the agent?

Comment (mikker) [reply Rakefile]: (This suggestion also needs an update to the indentation and one more `end` after the task.)

Comment (v1v) [reply Rakefile]: I actually used my web search skills and found an example for creating a task in rake, and unfortunately the example was with the `:environment`, thanks for the review :)

**Pull request merged**